api/src/main/java/org/openmrs/module/kenyaemr/calculation/library/hiv/GreenCardVelocityCalculation.java

```diff
@@ -79,20 +79,18 @@ public CalculationResultMap evaluate(Collection<Integer> cohort, Map<String, Obj
         CalculationResultMap tbCurrent = Calculations.lastObs(OnAntiTbQuestion, cohort, context);
         CalculationResultMap tbStarted = Calculations.lastObs(StartAntiTbQuestion, cohort, context);
 
-        //Check whether in ipt program
-        Concept IptCurrentQuestion = Context.getConceptService().getConcept(164949);
-        Concept IptStartQuestion = Context.getConceptService().getConcept(1265);
-        Concept IptStopQuestion = Context.getConceptService().getConcept(160433);
+        //Check whether completed 6 months ipt program
+        Concept IptCompletionQuestion = Context.getConceptService().getConcept(161555);
```

Comment:
this question has been used in multiple program discontinuation form
		public static final String REASON_FOR_PROGRAM_DISCONTINUATION = "161555AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA";
Kindly get the last ipt discontinuation encounter and fetch the obs for discontinuation reason. As it is, your code will pull even mch/tb/hiv discontinuation